addons/account/models/account_move.py

```diff
@@ -1242,18 +1245,22 @@ def _check_duplicate_supplier_reference(self):
             return
 
         self._cr.execute('''
-            SELECT move.id
-            FROM account_move move
+            WITH current_moves (id,  ref, company_id, commercial_partner_id, type, invoice_date) as ( values {} )
+            SELECT move2.id
+            FROM (    SELECT id, ref, company_id, commercial_partner_id, type, invoice_date from account_move
+            UNION ALL SELECT id, ref, company_id, commercial_partner_id, type, invoice_date from current_moves
+            ) move
             INNER JOIN account_move move2 ON
                 move2.ref = move.ref
                 AND move2.company_id = move.company_id
                 AND move2.commercial_partner_id = move.commercial_partner_id
                 AND move2.type = move.type
+                AND (move2.invoice_date = move.invoice_date OR move.invoice_date is NULL)
                 AND move2.id != move.id
             WHERE move.id IN %s
             AND move.type in ('in_invoice', 'in_refund')
             AND move.ref IS NOT NULL
-        ''', [tuple(self.ids)])
+        '''.format(", ".join(["(%s, %s, %s, %s, %s, CAST(%s as DATE))"] * len(self))), [i for list in self.mapped(lambda r: (r.id, r.ref, r.company_id.id, r.commercial_partner_id.id, r.type, r.invoice_date or None)) for i in list] + [tuple(moves.ids)])
```

Comment:
not easy enough to read. Please build the args in a dedicated loop

Reply:
is it  better with `itertools.chain` ?